### PR TITLE
migration asset selectors from lodash

### PR DESCRIPTION
**Pull request by terrysahaidak**
Fixes RNBW-3288

## What changed (plus any additional context for devs)

I removed lodash from assets selectors and directly used modules.

Before the changes, each execution on Samsung a12 2021 in release would take up to 200+ms.
After the change, it's ~120ms. After the react-native version bump and using the proper Hermes transform profile, I expect improvement to be even more significant because of native support for for in/of loops.

I used this simple function to track performance:

```js
export function trackPerf(fn, name) {
  return function wrapper() {
    const t = performance.now();

    const res = fn.apply(this, arguments);

    const total = performance.now() - t;
    
    global.console.log(`Event ${name}: ${total.toFixed(2)}ms`); // global is used to see logs in production

    return res;
  };
}
```

You can see all the logs in production using:
```sh
npx react-native log-android
```

## Dev checklist for QA: what to test

- wallet screen should show all the assets in the correct order

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [x] Added e2e tests? if not please specify why
- [x] If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/helpers/assetSelectors.js [hunk 1]
```diff
@@ -65,14 +66,20 @@ const sortAssetsByNativeAmount = (
...
+
+    return itemA < itemB ? 1 : -1;
+  });
+
+  const sortedShitcoins = noValue.sort((a, b) => {
+    let itemA = a.name;
+    let itemB = b.name;
+
+    return itemA > itemB ? 1 : -1;
+  });
```
Comment (osdnk) [inline src/helpers/assetSelectors.js:81]: I have always had a problem with "pretending" that this is immutable when this is not.

**Review by osdnk — APPROVED** [review]

**Review by brunobar79 — APPROVED** [review]
🔥🔥🔥  

PD: can u rebase so CI can run e2e properly?

**Review by jkadamczyk — APPROVED** [review]
LGTM

**Review by ibrahimtaveras00 — APPROVED** [review]
Checked against prod with different wallets, 👍🏽

**Comment by terrysahaidak** [comment]
Great! Thanks! Gonna fix the linting issues later today.

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**